runtime/kubernetes/client/kubernetes.go

```diff
@@ -1,43 +1,149 @@
+	// CreateService creates new kubernetes service
+	CreateService(*Service) error
+	// GetService queries kubernetes services and returns the matches
+	GetService(map[string]string) (*ServiceList, error)
+	// UpdateService patches kubernetes service
+	UpdateService(*Service) error
+	// DeleteService deletes kubernetes service
+	DeleteService(*Service) error
+	// ListServices lists all micro services running in Kubernetes
+	ListServices() (*ServiceList, error)
 }
 
-// Template is micro deployment template
-type Template struct {
-	Metadata *Metadata `json:"metadata,omitempty"`
-}
+// DefaultService returns default micro kubernetes service definition
+func DefaultService(name, version string) *Service {
+	Labels := map[string]string{
+		"name":    name,
+		"version": version,
+		"micro":   "service",
+	}
 
-// Spec defines micro deployment spec
-type Spec struct {
-	Template *Template `json:"template,omitempty"`
-}
+	svcName := name
+	if len(version) > 0 {
+		// API service object name joins name and version over "-"
+		svcName = strings.Join([]string{name, version}, "-")
 
-// Metadata defines api request metadata
-type Metadata struct {
-	Name        string            `json:"name,omitempty"`
-	Labels      map[string]string `json:"labels,omitempty"`
-	Annotations map[string]string `json:"annotations,omitempty"`
-}
+	}
 
-// DeploymentList
-type DeploymentList struct {
-	Items []Deployment `json:"items"`
-}
+	Metadata := &Metadata{
+		Name:      svcName,
+		Namespace: "default",
+		Version:   version,
+		Labels:    Labels,
+	}
 
-// Deployment is Kubernetes deployment
-type Deployment struct {
-	Metadata *Metadata `json:"metadata"`
-	Status   *Status   `json:"status"`
+	Spec := &ServiceSpec{
+		Type:     "ClusterIP",
+		Selector: Labels,
+		Ports: []ServicePort{{
+			name + "-port", 9090, "",
+		}},
+	}
+
+	return &Service{
+		Metadata: Metadata,
+		Spec:     Spec,
+	}
 }
 
-// Status is Kubernetes deployment status
-type Status struct {
-	Replicas          int `json:"replicas"`
-	AvailableReplicas int `json:"availablereplicas"`
+// DefaultService returns default micro kubernetes deployment definition
+func DefaultDeployment(name, version string) *Deployment {
+	Labels := map[string]string{
+		"name":    name,
+		"micro":   "service",
+		"version": version,
+	}
+
+	// API deployment object name joins name and version over "="
+	depName := strings.Join([]string{name, version}, "-")
+
+	Metadata := &Metadata{
+		Name:      depName,
+		Namespace: "default",
+		Version:   version,
+		Labels:    Labels,
+	}
+
+	// TODO: we need to figure out this version stuff; might need to add Build to runtime.Service
+	buildTime, err := strconv.ParseInt(version, 10, 64)
+	if err == nil {
+		buildUnixTimeUTC := time.Unix(buildTime, 0)
+		Metadata.Annotations = map[string]string{
+			"build": buildUnixTimeUTC.Format(time.RFC3339),
+		}
+	} else {
+		log.Debugf("Runtime could not parse build: %v", err)
+	}
+
+	Env := []EnvVar{
```

Comment:
We should leave this empty and then pass in the env configuration in the `micro runtime` service as something like

```
MICRO_REGISTRY=service
MICRO_BROKER=service
MICRO_REGISTRY_ADDRESS="registry-service"
MICRO_BROKER_ADDRESS="broker-service"
```